kairon/shared/data/collection_processor.py

```
@@ -0,0 +1,101 @@
+                if k not in is_non_editable
+            }
+
+            collection_obj.data.update(filtered_data)
+            collection_obj.collection_name = collection_name
+            collection_obj.is_secure = is_secure
+            collection_obj.is_non_editable = is_non_editable
+            collection_obj.user = user
+            collection_obj.timestamp = datetime.utcnow()
+            collection_obj.save()
+        except DoesNotExist:
+            raise AppException("Collection Data with given id and collection_name not found!")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)